### PR TITLE
Refactored File sharing code to improve the use of conditional compilation

### PR DESCRIPTION
**Pull request by prabhjot-msft**
# What
Refactored File sharing code to improve the use of conditional compilation

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/FileUploadCards.tsx [hunk 1]
```diff
@@ -100,3 +94,13 @@ export const _FileUploadCards = (props: FileUploadCardsProps): JSX.Element => {
     </_FileCardGroup>
   );
 };
+
+const useLocaleStringsTrampoline = (): _FileUploadCardsStrings => {
```
Comment (prprabhu-ms) [inline packages/react-components/src/components/FileUploadCards.tsx:98]: This function is repeated in two cards. Extract to common utility file?

File: packages/react-composites/src/composites/ChatComposite/file-sharing/FileUploadButton.tsx [hunk 2]
```diff
@@ -124,3 +117,10 @@ export const FileUploadButtonWrapper = (
     </>
   );
 };
+
+const UploadFileButtonStringTrampoline = (): string => {
+  // @conditional-compile-remove(file-sharing)
+  return useLocale().strings.chat.uploadFile;
+  // Return _some_ available icon, as the real icon is beta-only.
```
Comment (prprabhu-ms) [inline packages/react-composites/src/composites/ChatComposite/file-sharing/FileUploadButton.tsx:124]: nit: misleading comment

File: packages/react-composites/src/composites/ChatComposite/file-sharing/FileUploadButton.tsx [hunk 3]
```diff
@@ -124,3 +117,10 @@ export const FileUploadButtonWrapper = (
     </>
   );
 };
+
+const UploadFileButtonStringTrampoline = (): string => {
```
Comment (prprabhu-ms) [inline packages/react-composites/src/composites/ChatComposite/file-sharing/FileUploadButton.tsx:121]: `UploadFileButtonStringTrampoline` -> `uploadFileButtonStringTrampoline`

**Review by prprabhu-ms — APPROVED** [review]

File: change/@internal-react-components-dca2a298-483c-4263-a85b-00791ddb483f.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (anjulgarg) [inline change/@internal-react-components-dca2a298-483c-4263-a85b-00791ddb483f.json:2]: none

File: change/@internal-react-composites-b23c4821-a67e-4f5d-a7b1-c9b40d8452b6.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (anjulgarg) [inline change/@internal-react-composites-b23c4821-a67e-4f5d-a7b1-c9b40d8452b6.json:2]: none

**Review by anjulgarg — APPROVED** [review]

**Pull request merged**